apps/hash-ai-worker-ts/src/workflows.ts

```diff
@@ -24,69 +25,124 @@ const graphActivities = proxyActivities<
 export const inferEntities = (params: InferEntitiesCallerParams) =>
   aiActivities.inferEntitiesActivity(params);
 
-export const updateEntityEmbeddings = async (params: {
+type UpdateEntityEmbeddingsParams = {
   authentication: {
     actorId: AccountId;
   };
-  entity: Entity;
-}): Promise<CreateEmbeddingResponse.Usage> => {
-  const subgraph = await graphActivities.getEntityTypesByQuery({
-    authentication: params.authentication,
-    query: {
-      filter: {
-        equal: [
-          { path: ["versionedUrl"] },
-          { parameter: params.entity.metadata.entityTypeId },
-        ],
+} & (
+  | {
+      entities: Entity[];
+    }
+  | {
+      filter: Filter;
+    }
+);
+
+export const updateEntityEmbeddings = async (
+  params: UpdateEntityEmbeddingsParams,
+): Promise<CreateEmbeddingResponse.Usage> => {
+  const temporalAxes = {
+    pinned: {
+      axis: "transactionTime",
+      timestamp: null,
+    },
+    variable: {
+      axis: "decisionTime",
+      interval: {
+        start: null,
+        end: null,
       },
-      graphResolveDepths: {
-        inheritsFrom: { outgoing: 255 },
-        constrainsValuesOn: { outgoing: 0 },
-        constrainsPropertiesOn: { outgoing: 1 },
-        constrainsLinksOn: { outgoing: 0 },
-        constrainsLinkDestinationsOn: { outgoing: 0 },
-        isOfType: { outgoing: 0 },
-        hasLeftEntity: { incoming: 0, outgoing: 0 },
-        hasRightEntity: { incoming: 0, outgoing: 0 },
+    },
+  } as const;
+
+  let entities: Entity[];
+
+  if ("entities" in params) {
+    entities = params.entities;
+  } else {
+    const subgraph = await graphActivities.getEntitiesByQuery({
+      authentication: params.authentication,
+      query: {
+        filter: params.filter,
+        graphResolveDepths: {
+          inheritsFrom: { outgoing: 0 },
+          constrainsValuesOn: { outgoing: 0 },
+          constrainsPropertiesOn: { outgoing: 0 },
+          constrainsLinksOn: { outgoing: 0 },
+          constrainsLinkDestinationsOn: { outgoing: 0 },
+          isOfType: { outgoing: 0 },
+          hasLeftEntity: { incoming: 0, outgoing: 0 },
+          hasRightEntity: { incoming: 0, outgoing: 0 },
+        },
+        temporalAxes,
+        includeDrafts: false,
       },
-      temporalAxes: {
-        pinned: {
-          axis: "transactionTime",
-          timestamp: null,
+    });
+
+    entities = await graphActivities.getSubgraphEntities({
+      subgraph,
+    });
+  }
+
+  const usage: CreateEmbeddingResponse.Usage = {
+    prompt_tokens: 0,
+    total_tokens: 0,
+  };
+
+  for (const entity of entities) {
+    // TODO: The subgraph library does not have the required methods to do this client side so for simplicity we're
+    //       just making another request here. We should add the required methods to the library and do this client
+    //       side.
```

Comment:
I think you could do it with these two methods from the subgraph library:
```ts
const propertyTypeIds = getPropertyTypesReferencedByEntityType(
  subgraph, 
  entity.metadata.entityTypeId
);

const propertyTypes = propertyTypeIds.map(
  propertyTypeId => getPropertyTypeById(subgraph, propertyTypeId)
);
```

Reply:
In theory it should work, yes, but I don't think this will take inheritance in account

Reply:
(To be clear: I did not know about the first function, but I know it was implemented before we implemented inheritance)